### DISCUSSION TITLE
Issue with images rendering sometimes

### DISCUSSION THREAD

**Discussion by HansMellman** [posit-dev/great-tables#500] [Q&A]
I have been experiencing issues where images in my title do not render - when I run the same thing in my 'GT Test' notebook, they render perfectly, when I have them as a function in a module that I reference in another script they print out without the images as in the pictures below - any idea why this is happening and what I can do to prevent it? I'd done this so that I can automate the output of these GT tables rather than having to manually trigger them myself via notebook. 

If this is unclear then I can explain a little further.

Thanks in advance!

![AFC_TOY](https://github.com/user-attachments/assets/5fe0632a-a574-44f3-9793-a1bf6c11eeae)
![NFC_TOY](https://github.com/user-attachments/assets/23d5c5ae-2d05-4f78-b7e0-248e9e8fb2a4)

**Comment by jrycw** [comment]
I'm not sure of the root cause of this issue, but I'm curious if [val_fmt_image()](https://github.com/posit-dev/great-tables/pull/451) could help resolve it.

**Reply by HansMellman** [reply]
Thanks for the response, I'm wondering if it's something to do with the Save function and its utilisation of Chrome - as when I was trying to save this image yesterday I repeatedly had this issue when using a VPN connection that was slowing my internet down considerably - I disconnected and then re-tested my internet speed and I was able to successfully get the images to render, as they're stored in imgur links, not as local files.

**Reply by jrycw** [reply]
Thanks for sharing your experience. It seems that a sub-optimal connection speed might impact image rendering. That’s an interesting issue I hadn’t considered before.

**Reply by HansMellman** [reply]
That's the only significant change I made between attempts, I'd just thought of it because it's my understanding that to render the table it uses Chrome and then essentially takes a 'Snapshot' picture of it, right? before then saving it to a file - if that action happens before it has a chance to render the images included in the table (if they're hosted, not local) then I guess it could end up like that example. Though of course my understanding of the package is amateur at best and I don't know the exact order of operations behind the scenes in that particular function, so really it's just guess work!
